app/models/parser_version.rb

```diff
@@ -10,16 +10,16 @@ class ParserVersion < ActiveResource::Base
   self.element_name = 'version'
 
   def last_harvested_at
-    job = harvest_jobs('finished')[1]
-    job ? job.start_time : nil
-  end
-
-  def harvest_jobs(status = nil)
-    if status == 'finished'
-      HarvestJob.where(parser_id: parser_id, status: status).desc(:start_time)
-    else
-      HarvestJob.where(parser_id: parser_id).desc(:start_time)
-    end
+    last_finished_harvest_job = HarvestJob.desc(:start_time)
+      .find_by(
+      parser_id: parser_id,
+      status: 'finished',
+      environment: { '$ne': 'preview' }
+    )
+
+    last_finished_harvest_job.start_time
+  rescue Mongoid::Errors::DocumentNotFound
+    nil
```

Comment:
Can you throw a useful error here?

Reply:
It probably doesn't matter actually

Reply:
I just want to return nil if it was not found, do you know if there is a better method that doesn't throw an error and return just 1 or nil? haha

Reply:
Per the ActiveRecord docs, find_by should return nil if not found rather than raising an exception... however, in Mongoid... https://www.rubydoc.info/github/mongoid/mongoid/Mongoid%2FFindable:find_by
it relies on the `Mongoid.raise_not_found_error` variable, if it's set to `true` it raises an exception (which is our case 😞). I'm kinda reluctant to change that especially in the worker 😕

Reply:
Yeah fair enough, I think this is OK 👍